code/modules/admin/verbs/ghost_pool_protection.dm

```diff
@@ -0,0 +1,82 @@
+//very similar to centcom_podlauncher in terms of how this is coded, so i kept a lot of comments from it
+
+/client/proc/ghost_pool_protection() //Creates a verb for admins to open up the ui
+	set name = "Ghost Pool Protection"
+	set desc = "Choose which ways people can get into the round, or just clear it out completely for admin events."
+	set category = "Admin - Events"
+	var/datum/ghost_pool_menu/tgui  = new(usr)//create the datum
+	tgui.ui_interact(usr)//datum has a tgui component, here we open the window
+
+/datum/ghost_pool_menu
+	var/client/holder //client of whoever is using this datum
+
+	//when submitted, what the pool flags will be set to
+	var/new_role_flags = ALL
+
+	//EVERY TYPE OF WAY SOMEONE IS GETTING BACK INTO THE ROUND!
+	//these are the same comments as the ones in admin.dm defines, please update those if you change them here
+	/*
+	var/events_or_midrounds = TRUE //ie fugitives, space dragon, etc. also includes dynamic midrounds as it's the same deal
+	var/spawners = TRUE //ie ashwalkers, free golems, beach bums
+	var/station_sentience = TRUE //ie posibrains, mind monkeys, sentience potion, etc.
+	var/minigames = TRUE //ie mafia, ctf
+	var/misc = TRUE //oddities like split personality and any animal ones like spiders, xenos
+	*/
+
+/datum/ghost_pool_menu/New(H)//H can either be a client or a mob due to byondcode(tm)
+	if (istype(H,/client))
+		var/client/C = H
+		holder = C //if its a client, assign it to holder
+	else
+		var/mob/M = H
+		holder = M.client //if its a mob, assign the mob's client to holder
+	new_role_flags = GLOB.ghost_role_flags
+
+/datum/ghost_pool_menu/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/ghost_pool_menu/ui_close()
+	qdel(src)
+
+/datum/ghost_pool_menu/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "GhostPoolProtection")
+		ui.open()
+
+/datum/ghost_pool_menu/ui_data(mob/user)
+	var/list/data = list()
+	data["events_or_midrounds"] = (new_role_flags & GHOSTROLE_MIDROUND_EVENT)
+	data["spawners"] = (new_role_flags & GHOSTROLE_SPAWNER)
+	data["station_sentience"] = (new_role_flags & GHOSTROLE_STATION_SENTIENCE)
+	data["silicons"] = (new_role_flags & GHOSTROLE_SILICONS)
+	data["minigames"] = (new_role_flags & GHOSTROLE_MINIGAME)
+	return data
```

Comment:
```suggestion
	. = list()
	.["events_or_midrounds"] = (new_role_flags & GHOSTROLE_MIDROUND_EVENT)
	.["spawners"] = (new_role_flags & GHOSTROLE_SPAWNER)
	.["station_sentience"] = (new_role_flags & GHOSTROLE_STATION_SENTIENCE)
	.["silicons"] = (new_role_flags & GHOSTROLE_SILICONS)
	.["minigames"] = (new_role_flags & GHOSTROLE_MINIGAME)

```

Reply:
we ended up working it out on discord, data is much nicer